File: Java Script - Introduction to Programming/4.1. Complex Conditions/06. Point on Rectangle Border.js
function pointOnRectangleBorder([...arg]) {
  let [x1, y1, x2, y2, x, y] = [...arg.map(Number)]
  let findX = (x === x1 || x === x2) && (y >= y1) && (y <= y2);
  let findY = (y === y1 || y === y2) && (x >= x1) && (x <= x2);
  return findX || findY ? 'Border' : 'Inside / Outside'
}

//console.log(pointOnRectangleBorder(['2', '-3', '12', '3', '12', '-1']))

File: Pyrhon - Introduction to Programming/4.1. Complex Conditions/06. Point on Rectangle Border.py
x1, y1, x2, y2, x, y = [float(input()) for _ in range(6)]

findX = (x == x1 or x == x2) and (y >= y1) and (y <= y2)
findY = (y == y1 or y == y2) and (x >= x1) and (x <= x2)

print('Border' if findX or findY else 'Inside / Outside')


'''
2
-3
12
3
12
-1
#Border 
'''
